lib/mocker/express/server.js

```diff
@@ -73,6 +74,22 @@ class Server {
 
 			const uris = this._normalizeExpressPath(path.uri);
 
+			// Create a function that is memoized using the URL, query, the Prefer header and the body.
+			// eslint-disable-next-line no-unused-vars
+			const getResponse = (url, query, preferHeader, body) => {
+				const { example: preferredExampleName, statusCode: preferredStatusCode } = parsePreferHeader(preferHeader) || {};
+
+				if(preferredStatusCode)
+					logger.debug(`Searching requested response with status code ${preferredStatusCode}`);
+				else
+					logger.debug('Searching first response');
+				return path.getResponse(preferredStatusCode, preferredExampleName);
+			};
+
+			const getResponseMemo = memoize(getResponse, {
```

Comment:
This part of code I am not 100% sure about since the caching is done in-process.
This means that if the host ends up having to scale, repeated requests will come back with a new dynamic response depending on the instance that has been hit.

I don't know if you want this addressing now or whether you feel like an external caching strategy can be added later?